pyccel/parser/semantic.py

```diff
@@ -555,7 +572,8 @@ def _extract_indexed_from_var(self, var, indices, expr):
 
         if isinstance(var, PythonTuple):
             def is_literal_index(a):
-                is_int = lambda a: isinstance(a, (int, LiteralInteger)) or \
+                def is_int(a):
+                    return isinstance(a, (int, LiteralInteger)) or \
                         (isinstance(a, PyccelUnarySub) and \
                          isinstance(a.args[0], (int, LiteralInteger)))
```

Comment:
Does Pylint complain about the lambda function?

Reply:
Yes I was getting the error [unnecessary-lambda-assignment](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/unnecessary-lambda-assignment.html):

> pyccel/parser/semantic.py:573:25: C3001: Lambda expression assigned to a variable. Define a function using the "def" keyword instead. (unnecessary-lambda-assignment)

There is an explanation for why functions should be preferred [here](https://treyhunner.com/2018/09/stop-writing-lambda-expressions/#Misuse:_naming_lambda_expressions)